### ISSUE TITLE
[BUG]: Warning: Failed to Disable MLflow Autologging for Gluon

### ISSUE THREAD

**Issue by Saisri534** [zenml-io/zenml#3297]
### Contact Details [Optional]

saisrimalireddy@gmail.com

### System Information

ZENML_LOCAL_VERSION: 0.72.0
ZENML_SERVER_VERSION: 0.72.0
ZENML_SERVER_DATABASE: sqlite
ZENML_SERVER_DEPLOYMENT_TYPE: local
ZENML_CONFIG_DIR: /home/saisri/.config/zenml
ZENML_LOCAL_STORE_DIR: /home/saisri/.config/zenml/local_stores
ZENML_SERVER_URL: http://127.0.0.1:8237
ZENML_ACTIVE_REPOSITORY_ROOT: /home/saisri/mlops/datascience/backend/app
PYTHON_VERSION: 3.10.12
ENVIRONMENT: native
SYSTEM_INFO: {'os': 'linux', 'linux_distro': 'ubuntu', 'linux_distro_like': 'debian', 
'linux_distro_version': '22.04'}
ACTIVE_WORKSPACE: default
ACTIVE_STACK: test_stack_new
ACTIVE_USER: default
TELEMETRY_STATUS: enabled
ANALYTICS_CLIENT_ID: 406d7ac9-2472-49b2-8454-63f7159d583a
ANALYTICS_USER_ID: c6b8302a-2132-408f-9479-9db20b106d3f
ANALYTICS_SERVER_ID: 406d7ac9-2472-49b2-8454-63f7159d583a
INTEGRATIONS: ['airflow', 'bitbucket', 'numpy', 'pandas', 'kaniko', 'openai', 'pigeon', 'pillow', 
'pytorch', 'scipy']
PACKAGES: {'deprecated': '1.2.14', 'gitpython': '3.1.43', 'mako': '1.3.6', 'markdown': '3.7', 
'markupsafe': '3.0.2', 'pyjwt': '2.7.0', 'pymysql': '1.1.1', 'pyyaml': '6.0.1', 'restrictedpython': 
'7.4', 'sqlalchemy': '2.0.36', 'sqlalchemy-utils': '0.41.2', 'secretstorage': '3.3.3', 'absl-py': 
'2.1.0', 'accelerate': '1.1.0', 'aim': '3.27.0', 'aim-mlflow': '0.2.1', 'aim-ui': '3.27.0', 'aimrecords':
'0.0.7', 'aimrocks': '0.5.2', 'aiofiles': '24.1.0', 'aiohappyeyeballs': '2.4.3', 'aiohttp': '3.10.10', 
'aiosignal': '1.3.1', 'alembic': '1.8.1', 'annotated-types': '0.7.0', 'anyio': '4.6.2.post1', 
'asttokens': '2.4.1', 'astunparse': '1.6.3', 'async-timeout': '4.0.3', 'attrs': '24.2.0', 
'backports.tarfile': '1.2.0', 'base58': '2.0.1', 'bcrypt': '4.0.1', 'blinker': '1.8.2', 'boto3': 
'1.35.54', 'botocore': '1.35.54', 'cachetools': '5.5.0', 'certifi': '2024.8.30', 'cffi': '1.17.1', 
'charset-normalizer': '3.4.0', 'click': '8.1.3', 'cloudpickle': '2.2.1', 'colorama': '0.4.6', 'comm': 
'0.2.2', 'contourpy': '1.3.0', 'cryptography': '43.0.3', 'cycler': '0.12.1', 'databricks-sdk': '0.36.0', 
'dataclasses-json': '0.6.7', 'datasets': '3.1.0', 'decorator': '5.1.1', 'dill': '0.3.8', 'distro': 
'1.9.0', 'docker': '7.1.0', 'docstring-parser': '0.16', 'docutils': '0.21.2', 'entrypoints': '0.4', 
'evaluate': '0.4.3', 'exceptiongroup': '1.2.2', 'executing': '2.1.0', 'fastapi': '0.110.0', 'filelock': 
'3.16.1', 'flask': '3.0.3', 'flatbuffers': '24.3.25', 'fonttools': '4.54.1', 'frozenlist': '1.5.0', 
'fsspec': '2024.9.0', 'gast': '0.6.0', 'gitdb': '4.0.11', 'google-auth': '2.35.0', 'google-pasta': 
'0.2.0', 'graphene': '3.4.1', 'graphql-core': '3.2.5', 'graphql-relay': '3.2.0', 'greenlet': '3.1.1', 
'grpcio': '1.67.1', 'gunicorn': '22.0.0', 'h11': '0.14.0', 'h5py': '3.12.1', 'httpcore': '1.0.6', 
'httptools': '0.6.4', 'httpx': '0.27.2', 'huggingface-hub': '0.24.7', 'idna': '3.10', 
'importlib-metadata': '7.2.1', 'instructor': '1.5.2', 'interegular': '0.3.3', 'ipinfo': '5.1.1', 
'ipython': '8.29.0', 'ipywidgets': '8.1.5', 'itsdangerous': '2.2.0', 'jaraco.classes': '3.4.0', 
'jaraco.context': '6.0.1', 'jaraco.functools': '4.1.0', 'jedi': '0.19.1', 'jeepney': '0.8.0', 'jinja2': 
'3.1.4', 'jiter': '0.5.0', 'jmespath': '1.0.1', 'joblib': '1.4.2', 'jsonpatch': '1.33', 'jsonpointer': 
'3.0.0', 'jupyterlab-widgets': '3.0.13', 'keras': '3.5.0', 'keyring': '25.5.0', 'kiwisolver': '1.4.7', 
'langchain': '0.2.0', 'langchain-community': '0.2.4', 'langchain-core': '0.2.43', 
'langchain-text-splitters': '0.2.4', 'langsmith': '0.1.140', 'libclang': '18.1.1', 'lm-format-enforcer': 
'0.10.7', 'loglizer': '1.0', 'logparser3': '1.0.4', 'markdown-it-py': '3.0.0', 'marshmallow': '3.23.1', 
'matplotlib': '3.9.2', 'matplotlib-inline': '0.1.7', 'mdurl': '0.1.2', 'ml-dtypes': '0.4.1', 'mlflow': 
'2.20.0rc0', 'mlflow-skinny': '2.20.0rc0', 'more-itertools': '10.5.0', 'mpmath': '1.3.0', 'multidict': 
'6.1.0', 'multiprocess': '0.70.16', 'mypy-extensions': '1.0.0', 'namex': '0.0.8', 'networkx': '3.4.2', 
'nh3': '0.2.18', 'numpy': '1.26.4', 'nvidia-cublas-cu12': '12.4.5.8', 'nvidia-cuda-cupti-cu12': 
'12.4.127', 'nvidia-cuda-nvrtc-cu12': '12.4.127', 'nvidia-cuda-runtime-cu12': '12.4.127', 
'nvidia-cudnn-cu12': '9.1.0.70', 'nvidia-cufft-cu12': '11.2.1.3', 'nvidia-curand-cu12': '10.3.5.147', 
'nvidia-cusolver-cu12': '11.6.1.9', 'nvidia-cusparse-cu12': '12.3.1.170', 'nvidia-nccl-cu12': '2.21.5', 
'nvidia-nvjitlink-cu12': '12.4.127', 'nvidia-nvtx-cu12': '12.4.127', 'oauthlib': '3.2.2', 'openai': 
'1.54.1', 'opentelemetry-api': '1.28.0', 'opentelemetry-sdk': '1.28.0', 
'opentelemetry-semantic-conventions': '0.49b0', 'opt-einsum': '3.4.0', 'optree': '0.13.0', 'orjson': 
'3.10.11', 'packaging': '24.1', 'pandas': '2.2.3', 'parso': '0.8.4', 'passlib': '1.7.4', 'pexpect': 
'4.9.0', 'pillow': '11.0.0', 'pip': '24.3.1', 'pkginfo': '1.10.0', 'prompt-toolkit': '3.0.48', 
'propcache': '0.2.0', 'protobuf': '4.25.5', 'psutil': '6.1.0', 'ptyprocess': '0.7.0', 'pure-eval': 
'0.2.3', 'pyarrow': '15.0.2', 'pyasn1': '0.6.1', 'pyasn1-modules': '0.4.1', 'pycparser': '2.22', 
'pydantic': '2.8.2', 'pydantic-core': '2.20.1', 'pydantic-settings': '2.6.1', 'pygments': '2.18.0', 
'pyparsing': '3.2.0', 'pyphen': '0.17.0', 'python-dateutil': '2.9.0.post0', 'python-dotenv': '1.0.1', 
'python-multipart': '0.0.17', 'pytz': '2024.2', 'querystring-parser': '1.2.4', 'readme-renderer': '44.0',
'regex': '2022.3.2', 'requests': '2.32.3', 'requests-oauthlib': '2.0.0', 'requests-toolbelt': '1.0.0', 
'rfc3986': '2.0.0', 'rich': '13.9.4', 'rsa': '4.9', 's3transfer': '0.10.3', 'safetensors': '0.4.5', 
'scikit-learn': '1.5.2', 'scipy': '1.14.1', 'secure': '0.3.0', 'setuptools': '75.3.0', 'shellingham': 
'1.5.4', 'six': '1.16.0', 'smmap': '5.0.1', 'sniffio': '1.3.1', 'sqlmodel': '0.0.18', 'sqlparse': 
'0.5.1', 'stack-data': '0.6.3', 'starlette': '0.36.3', 'sympy': '1.13.1', 'tenacity': '8.5.0', 
'tensorboard': '2.17.1', 'tensorboard-data-server': '0.7.2', 'tensorflow': '2.17.0', 
'tensorflow-io-gcs-filesystem': '0.37.1', 'termcolor': '1.1.0', 'textstat': '0.7.4', 'threadpoolctl': 
'3.5.0', 'tiktoken': '0.8.0', 'tokenizers': '0.19.1', 'torch': '2.5.1', 'tqdm': '4.66.6', 'traitlets': 
'5.14.3', 'transformers': '4.41.2', 'triton': '3.1.0', 'twine': '5.1.1', 'typer': '0.12.5', 
'typing-extensions': '4.12.2', 'typing-inspect': '0.9.0', 'tzdata': '2024.2', 'urllib3': '2.2.3', 
'uvicorn': '0.32.0', 'uvloop': '0.21.0', 'watchfiles': '0.24.0', 'wcwidth': '0.2.13', 'websockets': 
'13.1', 'werkzeug': '3.1.2', 'wheel': '0.44.0', 'widgetsnbextension': '4.0.13', 'wrapt': '1.16.0', 
'xxhash': '3.5.0', 'yarl': '1.17.1', 'zenml': '0.72.0', 'zipp': '3.20.2', 'autocommand': '2.2.2', 
'importlib-resources': '6.4.0', 'inflect': '7.3.1', 'jaraco.collections': '5.1.0', 'jaraco.text': 
'3.12.1', 'platformdirs': '4.2.2', 'tomli': '2.0.1', 'typeguard': '4.3.0'}

CURRENT STACK

Name: test_stack_new
ID: 1dcedeb1-111d-4b7f-8283-770fcc5a17b1
User: default / c6b8302a-2132-408f-9479-9db20b106d3f
Workspace: default / 6192e5bb-4bc8-46b0-9bbb-974345d83ea0

ORCHESTRATOR: default

Name: default
ID: 1c1b12d0-d100-4b25-bb2b-368ecd0d9646
Type: orchestrator
Flavor: local
Configuration: {}
Workspace: default / 6192e5bb-4bc8-46b0-9bbb-974345d83ea0

ARTIFACT_STORE: default

Name: default
ID: 0ee69cfe-bf08-496f-a1d4-4129e3622ed1
Type: artifact_store
Flavor: local
Configuration: {'path': ''}
Workspace: default / 6192e5bb-4bc8-46b0-9bbb-974345d83ea0

EXPERIMENT_TRACKER: mlflow_new_experiments_tracker

Name: mlflow_new_experiments_tracker
ID: e10e4067-54f1-4e59-accf-6ddede2264ce
Type: experiment_tracker
Flavor: mlflow
Configuration: {'experiment_name': None, 'nested': False, 'tags': {}, 'tracking_uri': 
'http://127.0.0.1:5000', 'tracking_username': '********', 'tracking_password': '********', 
'tracking_token': '********', 'tracking_insecure_tls': False, 'databricks_host': None, 
'enable_unity_catalog': False}
User: default / c6b8302a-2132-408f-9479-9db20b106d3f
Workspace: default / 6192e5bb-4bc8-46b0-9bbb-974345d83ea0

MODEL_REGISTRY: register_mlflow_models

Name: register_mlflow_models
ID: afdc0144-040e-4948-acc6-c16ae59ad14e
Type: model_registry
Flavor: mlflow
Configuration: {}
User: default / c6b8302a-2132-408f-9479-9db20b106d3f
Workspace: default / 6192e5bb-4bc8-46b0-9bbb-974345d83ea0

MLflow version: 2.20.0rc0
Python: 3.10.12
Operating System:

Host OS: Windows 10 
WSL Distribution: Ubuntu (Ubuntu 22.04)

### What happened?

While running the pipeline with the experiment tracker using the MLflow flavor, the following warning is displayed:
```
Failed to disable MLflow autologging for the following frameworks: ['gluon'].
```
This issue seems to be caused by the removal of the mlflow.gluon.autolog method in the latest version of MLflow (2.20.0rc0). 
mlflow.gluon.autolog is depicrated after the version 2.5.0
In the newer version of mlflow(2.20.0rc0) they have removed this method.

### Reproduction steps

1.Use the experiment tracker with MLflow flavor in a pipeline.
2.Run the pipeline with MLflow version 2.20.0rc0.
3.Observe the warning regarding Gluon autologging.
...

### Relevant log output

```shell
Initiating a new run for the pipeline: lstm_training.
Caching is disabled by default for lstm_training.
Using user: default
Using stack: test_stack_new
  artifact_store: default
  orchestrator: default
  model_registry: register_mlflow_models
  experiment_tracker: mlflow_new_experiments_tracker
Dashboard URL for Pipeline Run: http://127.0.0.1:8237/runs/74ca8e46-ec93-4bb4-ac23-7389241e9088
Step ingest_data has started.
Failed to disable MLflow autologging for the following frameworks: ['gluon'].
🏃 View run lstm_training-2025_01_20-11_20_52_940785 at: http://127.0.0.1:5000/#/experiments/248254685622856610/runs/9fd752ff919c4c269640a55e9ade0b02
🧪 View experiment at: http://127.0.0.1:5000/#/experiments/248254685622856610
Step ingest_data has finished in 8.546s.
Step clean_data has started.
Failed to disable MLflow autologging for the following frameworks: ['gluon'].
🏃 View run lstm_training-2025_01_20-11_20_52_940785 at: http://127.0.0.1:5000/#/experiments/248254685622856610/runs/9fd752ff919c4c269640a55e9ade0b02
🧪 View experiment at: http://127.0.0.1:5000/#/experiments/248254685622856610
Step clean_data has finished in 1.198s.
Step preprocess_data has started.
Failed to disable MLflow autologging for the following frameworks: ['gluon'].
```

### Code of Conduct

- [x] I agree to follow this project's Code of Conduct

**Comment by htahir1** [comment]
Fixed by #3298
